hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java

```diff
@@ -426,18 +432,20 @@ public int getStaleNodeCount() {
    * @return dead node count
    */
   public int getDeadNodeCount() {
-    return getNodeCount(NodeState.DEAD);
+    // TODO - hard coded IN_SERVICE
+    return getNodeCount(
+        new NodeStatus(NodeOperationalState.IN_SERVICE, NodeState.DEAD));
```

Comment:
Interesting; what happens to a node in maintenance mode, but switched off? or dead? Does that become a dead node? I think I agree with your conclusion that it is not a dead node, but flagging for others to consider.

Reply:
There are a bunch of places where I have hardcoded IN_SERVICE, so once we get this working we will need different events for DECOM / IN_MAINT + DEAD, as that is an expected state rather than an error condition as it would be now.

Reply:
Perfect, works well. I saw that later in the code. It is fine for now.